flocker/node/agents/functional/test_cinder_behaviour.py

```diff
@@ -0,0 +1,84 @@
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Tests for cinder API behaviour.
+"""
+
+from twisted.trial.unittest import SynchronousTestCase
+
+from ..cinder import wait_for_volume
+from ..testtools import tidy_cinder_client_for_test
+from ....testtools import random_name
+
+
+class VolumesCreateTests(SynchronousTestCase):
+    """
+    Tests for ``cinder.Client.volumes.create``.
+    """
+    def setUp(self):
+        self.cinder_client = tidy_cinder_client_for_test(test_case=self)
+
+    def test_create_metadata_is_listed(self):
+        """
+        ``metadata`` supplied when creating a volume is included when that
+        volume is subsequently listed.
+        """
+        expected_metadata = {random_name(): "bar"}
+
+        new_volume = self.cinder_client.volumes.create(
+            size=100,
+            metadata=expected_metadata
+        )
+        listed_volume = wait_for_volume(
+            volume_manager=self.cinder_client.volumes,
+            expected_volume=new_volume,
+        )
+
+        expected_items = set(expected_metadata.items())
+        actual_items = set(listed_volume.metadata.items())
+        missing_items = expected_items - actual_items
+        self.assertEqual(
+            set(), missing_items,
+            'Metadata {!r} does not contain the expected items {!r}'.format(
+                actual_items, expected_items
+            )
+        )
+    test_create_metadata_is_listed.todo = (
```

Comment:
They report that it's a `driver` bug.
I'm not sure if that's something specific to Rackspace but yes, I think I will want to run this against other OpenStacks. The client_from_environment thing should make that reasonably easy to do.